src/server/game/Entities/Pet/Pet.cpp

```diff
@@ -860,13 +860,23 @@ bool Guardian::InitStatsForLevel(uint8 petlevel)
     for (uint8 i = SPELL_SCHOOL_HOLY; i < MAX_SPELL_SCHOOL; ++i)
         SetModifierValue(UnitMods(UNIT_MOD_RESISTANCE_START + i), BASE_VALUE, float(cinfo->resistance[i]));
 
-    //health, mana, armor and resistance
+    //health, mana/power, armor and resistance
     PetLevelInfo const* pInfo = sObjectMgr->GetPetLevelInfo(creature_ID, petlevel);
     if (pInfo)                                      // exist in DB
     {
         SetCreateHealth(pInfo->health);
-        if (petType != HUNTER_PET) //hunter pet use focus
-            SetCreateMana(pInfo->mana);
+
+        // Choose power type based on unit_class, hunter pet already set to focus 
+        if (petType != HUNTER_PET) switch (cinfo->unit_class) 
```

Comment:
afaik this is not a TC code standard, you should creature a new line for the switch

Reply:
Yes, please move the
```cpp
switch (cinfo->unit_class)
```
part down to a new line with correct indent.
(look up existing  `switch` sections in TC code if you are uncertain)